### ISSUE TITLE
New page content template with upper-right image and list of links below

### ISSUE THREAD

**Issue by CrispinF** [jncc/chm-website#46]
1. A single image with alt text. This to be rendered upper right, right-aligned, with some left margin.  This will look best when moving between pages if the images are always the same width; we can enforce a max-width if you'd like - please specify the width in pixels if so. Or we could leave it unconstrained so it's up to you to pre-size the images carefully. Please confirm.
Question: would you like a visible caption below the image (which could be the same as the alt text)?

2. Manually-curated list of links, each with link address and text to show. Render with a standard title "Relevant Links" (to be confirmed) but only show the heading if there are any links configured.
Links to open in new window/tab (each with standard tooltip stating this).

On smaller screens, image and list of links to fall below the main content.

**Comment by johawker** [comment]
1. Visible caption not necessary @jonparsonsjncc to confirm regarding pixels and width

2. Standard title 'Relevant Links' is great, thanks

**Comment by CrispinF** [comment]
@johawker  I've converted one of your existing pages into how we might render it with a page template. Please review https://chm-staging.esdm.co.uk/uk-reports-to-cbd 
As noted above, if you enter no links the "Relevant Links" section would disappear.

**Comment by johawker** [comment]
@CrispinF I like the format of the converted page, how would I do this for the other pages? or is it that you'll need to edit the template for all? I tried to make a new page like this and couldn't figure out how to do it. I think it looks good and would like to apply it to the rest of the pages, thanks!

**Comment by CrispinF** [comment]
@johawker in the absence of template, the best way to do this is to copy out the source html of the one I did into a text editor such as Notepad++, review the page you are wanting to modify and copy the relevant texts into your text editor version, and then use the result to replace the existing html for the page you are working on. Does that make sense?
Once you have the structure in place, you can then probably further edit the main text and links etc in the WYSIWYG editor.
The structure is achieved mostly with the enclosing div elements.

**Comment by johawker** [comment]
@CrispinF haha not really although it might do to @jonparsonsjncc, happy to wait for a template anyway will be easier for future users. Thanks for all your help!

**Comment by CrispinF** [comment]
I'm not sure I can have a template ready for a few days.  It's quite simple - I'll explain how to covert an existing page...

Working in the source code view, wrap the entire lot in this
`<div class="d-flex row">`
... all the html you already had...
`</div>`

Next extract the image bit from the main text, and put it down the bottom before the final `<div>` that you just added.

Next enclose that image bit with this:
`<div class="col-md-4">`
... your image bit ...
`</div>`

Next wrap the wordy bit (which might be several paragraphs) in this
`<div class="col-md-8">`
... wordy stuff...
`</div>`

The top should now look like this:
```
<div class="d-flex row">
<div class="col-md-8">
```
...

Next modify your image so it's like this, i.e. wrap in figure, delete any style, add the figcaption:

```
<figure>
<img alt="image of a person writing on a clipboard in the countryside" src="/shutterstock_662344828-ws.jpg" />
<figcaption>Person writing. Credits: blah</figcaption>
</figure>
```

Finally, if you have any links, add and modify a section like this immediately after the figure:
```
<h3>Relevant Links</h3>
<ul>
	<li><a target="_blank" title="Open the CBD website (new window)" href="https://www.cbd.int/reports/search/">The CBD website</a></li>
	<li><a target="_blank" title="Open Article 26 (new window)" href="https://www.cbd.int/convention/articles/?a=cbd-26">Article 26</a></li>
</ul>
```

The reason I'd sometimes do this in a tool like Notepad++ is that the website editor will try to fix any mistakes when you save, and that can sometimes do more damage than help, plus it tends to change the indentation/spacing, so I like to work on it in Notepad++ and paste the result in and save a draft >> review the result.

**Comment by johawker** [comment]
@CrispinF  thanks for the info but happy to wait for the template for now

**Comment by CrispinF** [comment]
@johawker @jonparsonsjncc I've deployed the template to the staging site
Here's an example I made with it https://chm-staging.esdm.co.uk/chm-test8 
I've made the output able to work with or without an image and with or without links. So you can have a content-only page or content plus image, content plus links, or all three.
Have a go.
Once your happy, we can deploy this to the live site and you can remake all pages.
NB for the images, I've done it so you simply paste in the URL. There's no image uploader etc - it assumes you have done that already using the file manager etc. You can get the URL for an image on one of your existing pages by right-click on it / insteact html sources, etc. Use relative addresses for images without the site itself.
I would suggest trying to use images that have the same dimensions and aspect ratio, for a tidy result across pages.

**Comment by johawker** [comment]
@CrispinF I've tested this on the staging site and seems to work fine, thanks, feel free to deploy it to the live site and I'll remake the pages - I need to transfer slugs accross when I publish the new pages right in order to keep the redirects working- i remember you saying something about this but can't find the comment! Thanks

**Comment by CrispinF** [comment]
@johawker I've deployed to the live site and checked the template works.

I'd suggest the process for replacing a page is as follows. NB in testing and refining this, I've mostly done https://uk.chm-cbd.net/uk-and-the-cbd for you, but you will want to add more links I expect.  Note that you can still choose whether or not to show the "Child Menu". I left it on for now on that page, but perhaps it isn't needed if you use the Links. I also confess at one point I messed up and floated lots of pages into the top menu! I hope it's all back to where it should be!

1. Navigate to the parent of the page to be replaced. If replacing a root-level page, start on the home page. Create the new page using the template, giving it the correct page heading. It will automatically generate a slug slightly different from the real one, as it avoids duplicates. It will normally add an extra "-" on the end.
2. Add your content/image/links, saving as Draft. This means only you can see the page. Check that your new page appears where it should in the menu hierarchy, and if it doesn't, alter the parent slug until it does.
3. Navigate to the page you are replacing, go into edit mode, take note of the slug (copy it to clipboard/notepad), and then alter the slug to something else (e.g. add "-old" to the end). NB if this page had child pages, there _might_ be temporary mayhem at this point, as those pages will now have no valid parent and might jump into the root menu, or they might travel with the re-slugged page due to caching (see below) - don't worry!
4. Change the slug of your draft to the slug the original page had, and publish. Any children might now be back in place, but they might still be clinging to their original parent... this is because the menu structure gets cached to some extent to improve performance. If the child pages don't magically jump into place, go to each of the immediate children, edit and publish (without making any changes) and that should bounce them into place.
5. Go to the old page, which should now have no children, and edit > delete.

I would work in two separate browser windows to make the transition quick. Also this helps with transferring content.
If you want to copy and paste content across, it's usually better to copy the html course than the final result, but you'll need to remove the image, any empty paragraphs, etc.

Hope this helps!

**Comment by johawker** [comment]
@CrispinF I think I have now updated all the webpages to the latest template and published these, I left the relevant links section off the UK and CBD page for now and kept the child menu, the UK implementation of CBD has both for now but anyway they are all in the templates and the redirects should still work as I updated the slugs! 

Thanks for all the advice, was extremely helpful especially the bit about children pages wondering as they did wonder a little bit.
I think this is all complete now! 

Thanks very much for the help.
